cobc/cobc.c

```diff
@@ -9427,15 +9601,7 @@ main (int argc, char **argv)
 	}
 
 	/* Output dependency list */
-	if (cb_depend_file) {
-		struct cb_text_list	*l;
-		fprintf (cb_depend_file, "%s: \\\n", cb_depend_target);
-		for (l = cb_depend_list; l; l = l->next) {
-			fprintf (cb_depend_file, " %s%s\n", l->text, l->next ? " \\" : "\n");
-		}
-		for (l = cb_depend_list; l; l = l->next) {
-			fprintf (cb_depend_file, "%s:\n", l->text);
-		}
+	if (cb_depend_file && cb_depend_file != stdout){
```

Comment:
stdout cannot be null, so the second part of that check is enough

Reply:
But `cb_depend_file` can be, and we do not want to "close" it if it is.